### DISCUSSION TITLE
Using `pipx` to install Poetry might not be the best idea

### DISCUSSION THREAD

**Discussion by temeddix** [python-poetry/poetry#8687] [Ideas]
During the past couple of months, I've found that the recommended installation method has changed from **Poetry installation script** to `pipx`. This means that we have to install `pipx` first to use Poetry.

The main question arises here - Why do we have to install another package manager(`pipx`), to install a package manager(Poetry)?

Of course, `pipx` is a bit different from package managers because it's focused on globally executable scripts. But the point is that we could provide a simpler way to install Poetry on the system. It would be nice if we don't need to install other installer first.

What if, we can make users install Poetry like this?
```bash
pip install poetry
python -m poetry ensurepath  # Just like pipx, add Poetry to PATH automatically. Needs a PR.
```

There's a logic for `ensurepath` in the `pipx` codebase. Since `pipx` is based on MIT license, we can refer to it, I believe.

- https://github.com/pypa/pipx/blob/aa5e9405c927e9a3a2455b0deab2282f007e342b/src/pipx/commands/ensure_path.py#L100-L155

If this installation method is acceptable, I will be happy to make a PR.

**Comment by radoering** [comment]
> Why do we have to install another package manager(`pipx`), to install a package manager(Poetry)?

We don't have to do it, it's just a simple and robust method to install Poetry. We made it the first recommendation because it's less error-prone than the installation script and simpler than the manual method. Nevertheless, the other methods are not deprecated, you can still use them.

> What if, we can make users install Poetry like this?

A simple pip install is not recommended. You should only do this if you know exactly what you are doing. Thereby, it's not a good default method. See the big red warning at the beginning of https://python-poetry.org/docs/#installation

The [manual method](https://python-poetry.org/docs/#installing-manually) shows how to use pip correctly (first create a venv). `ensurepath` could make sense for the manual method, but I suppose adding Poetry manually to the path is sufficient especially for this installation method.

**Comment by gsemet** [comment]
Hello
Here is why it is a very good idea to install poetry with pipx.
- I have n different python projects.
- Some are updated to the latest version of poetry, some are not (less maintained)
- they all have a weekly build that checks if they still build
- i want my buildsystem to be as isolated as possible, but some runner (windows) are shared
- on shared runner, if one job install poetry locally (pip install --user), this will affect another one. So if 2 builds starts at the same time and if they are using different version of poetry, they will conflict.
- to avoid that, I run poetry with

```bash
pipx run --spec "poetry==1.8.0" poetry
```

it is not ideal (it still perform a network connectivity for nothing), but I can have several jobs starting at the same time on my shared runner and each will use the right version of poetry independantly.

**Reply by abn** [reply]
The suffix option is also great. 

```sh
pipx run --suffix=@1.8 --spec "poetry==1.8.0" poetry
```
